src/auspex/analysis/io.py

```diff
@@ -12,7 +12,52 @@
 import h5py
 import numpy as np
 
-def load_from_HDF5(filename_or_fileobject):
+def load_from_HDF5(filename_or_fileobject, return_structured_array=True):
+    data = {}
+    if isinstance(filename_or_fileobject, h5py.File):
+        f = filename_or_fileobject
+    else:
+        f = h5py.File(filename_or_fileobject, 'r')
+    for groupname in f:
+        # Reconstruct the descrciptor
+        descriptor = DataStreamDescriptor()
+        g = f[groupname]
+        axis_refs = g['descriptor']
+        for ref in reversed(axis_refs):
+            ax = g[ref]
+            if ax.attrs['unstructured']:
+                # The entry for the main unstructured axis contains
+                # references to the constituent axes
+                points = np.vstack([g[e] for e in ax[:]]).T
```

Comment:
Points are expected as tuples coordinates in the form [(x1, y1), (x2, y2), ...]. There's probably a more elegant way of doing this.